packages/plugin-ext/src/main/browser/text-editor-model-service.ts

```diff
@@ -108,4 +110,7 @@ export class EditorModelServiceImpl implements EditorModelService {
         return results.reduce((a, b) => a && b, true);
     }
 
+    async createModelReference(uri: URI): Promise<MonacoEditorModel | undefined> {
+        return this.monacoModelService.createModelReference(uri).then(ref => ref.object);
```

Comment:
if we never disposes `ref` when model will be kept in the memory forever. 

From VS Code docs:
> Note that the lifecycle of the returned document is owned by the editor and not by the extension. That means an onDidClose-event can occur at any time after opening it.

It seems that they dispose model when all editors are closed, so i think we have to keep a ref till there is at least one editor is opened for it. When an editor for a model is opened then we dispose a ref in order to allow a model be garbage collected as soon as editor is closed.

It would be good to have a simple VS Code extension which just open a document and then check with a profiler that a model is destroyed then one opens and closes the editor for it.

Reply:
Also my assumptions should be verified against VS Code, maybe it just leaks models and extensions relying on it.